modules/test/playwright/fixtures/isolatedLayoutTest.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {mergeTests} from '@playwright/test';
+
+import {ApiHelpers} from '../helpers/ApiHelpers';
+import {Layout} from '../helpers/json-web-services/JSONWebServicesLayoutApiHelper';
+import getRandomString from '../utils/getRandomString';
+import {backendPageTest} from './backendPageTest';
+
+export interface IsolatedLayoutOptions {
+
+	/**
+	 * Whether to publish the layout if its type is 'content' (otherwise it must be left
+	 * `undefined`)
+	 */
+	publish?: boolean;
+
+	/**
+	 * Layout type (eg: 'content', 'portlet', ...)
+	 */
+	type?: string;
+}
+
+export interface IsolatedLayout {
+	layout: Layout;
+}
+
+const test = mergeTests(backendPageTest);
+
+function fillDefaultValues(options: IsolatedLayoutOptions) {
+	if (options.type === undefined) {
+		options.type = 'content';
+	}
+
+	if (options.publish === undefined && options.type === 'content') {
+		options.publish = true;
+	}
+}
+
+function isolatedLayoutTest(options: IsolatedLayoutOptions = {}) {
+	fillDefaultValues(options);
+
+	const fixtureImpl = test.extend<IsolatedLayout>({
+		layout: [
+			async ({backendPage}, use) => {
+				const apiHelpers = new ApiHelpers(backendPage);
+
+				let layout: Layout;
+
+				try {
+
+					// Get group 'guest'
+
+					const company =
+						await apiHelpers.jsonWebServicesCompany.getCompanyByWebId(
+							'liferay.com'
+						);
```

Comment:
Do we maybe want to create [an isolated site](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/fixtures/isolatedSiteTest.ts), as a part of isolated page creation?

Reply:
I thought about it but didn't need it for these tests. 

Additionally I don't see why you would need an isolated layout inside an isolated site. If the layout is isolated the site doesn't matter and if you request an isolated site you are the owner of the site, so you can do whatever you want there: create one, two, or a million isolated sites.

So, until it is absolutely needed I'd say we wouldn't do it. :shrug: